File: app/src/main/java/com/example/mypc/fastfoodfinder/model/Routing/MapCoordination.java
package com.example.mypc.fastfoodfinder.model.Routing;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.android.gms.maps.model.LatLng;
import com.google.gson.annotations.SerializedName;

/**
 * Created by nhoxb on 11/11/2016.
 */
public class MapCoordination  implements Parcelable{
    protected MapCoordination(Parcel in) {
        latitude = in.readDouble();
        longitude = in.readDouble();
    }

    public static final Creator<MapCoordination> CREATOR = new Creator<MapCoordination>() {
        @Override
        public MapCoordination createFromParcel(Parcel in) {
            return new MapCoordination(in);
        }

        @Override
        public MapCoordination[] newArray(int size) {
            return new MapCoordination[size];
        }
    };

    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public LatLng getLocation()
    {
        return new LatLng(latitude, longitude);
    }

    @SerializedName("lat")
    double latitude;
    @SerializedName("lng")
    double longitude;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeDouble(latitude);
        parcel.writeDouble(longitude);
    }
}

File: app/src/main/kotlin/com/iceteaviet/fastfoodfinder/data/remote/routing/model/MapCoordination.kt
package com.iceteaviet.fastfoodfinder.data.remote.routing.model

import android.os.Parcel
import android.os.Parcelable

import com.google.android.gms.maps.model.LatLng
import com.google.gson.annotations.SerializedName

/**
 * Created by Genius Doan on 11/11/2016.
 */
class MapCoordination constructor(`in`: Parcel) : Parcelable {
    @SerializedName("lat")
    val latitude: Double
    @SerializedName("lng")
    val longitude: Double

    val location: LatLng
        get() = LatLng(latitude, longitude)

    init {
        latitude = `in`.readDouble()
        longitude = `in`.readDouble()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeDouble(latitude)
        parcel.writeDouble(longitude)
    }

    companion object CREATOR : Parcelable.Creator<MapCoordination> {
        override fun createFromParcel(parcel: Parcel): MapCoordination {
            return MapCoordination(parcel)
        }

        override fun newArray(size: Int): Array<MapCoordination?> {
            return arrayOfNulls(size)
        }
    }
}
